base/uk.ac.stfc.isis.ibex.ui.logplotter/src/uk/ac/stfc/isis/ibex/ui/logplotter/LogPlotterHistoryPresenter.java

```diff
@@ -19,15 +19,109 @@
 
 package uk.ac.stfc.isis.ibex.ui.logplotter;
 
+import java.util.ArrayList;
+
+import org.csstudio.trends.databrowser2.Messages;
+import org.csstudio.trends.databrowser2.editor.DataBrowserEditor;
+import org.csstudio.trends.databrowser2.model.AxisConfig;
+import org.csstudio.trends.databrowser2.model.Model;
+import org.csstudio.trends.databrowser2.model.PVItem;
+import org.csstudio.trends.databrowser2.preferences.Preferences;
+import org.csstudio.ui.util.EmptyEditorInput;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IEditorReference;
+import org.eclipse.ui.PlatformUI;
+
+import uk.ac.stfc.isis.ibex.ui.UI;
 import uk.ac.stfc.isis.ibex.ui.blocks.presentation.PVHistoryPresenter;
 
+/**
+ * The class that is responsible for displaying the log plotter.
+ */
 public class LogPlotterHistoryPresenter implements PVHistoryPresenter {
-
-	private final LogPlotterDisplay display = new LogPlotterDisplay();
 	
+	/**
+	 * @return An arraylist of all the current databrowsers.
+	 */
+	private ArrayList<DataBrowserEditor> getCurrentDataBrowsers() {
+		ArrayList<DataBrowserEditor> dataBrowserEditors = new ArrayList<>();
+		
+		IEditorReference[] editorRefs = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getEditorReferences();
+		
+		for (IEditorReference editorRef : editorRefs) {
+			IEditorPart editor = editorRef.getEditor(false);
+			if (editor instanceof DataBrowserEditor) {
+				dataBrowserEditors.add((DataBrowserEditor) editor);
+			}
+		}
+		return dataBrowserEditors;
+	}
+	
+	@Override
+	public ArrayList<String> getCurrentPresenters() {
+		ArrayList<String> editorNames = new ArrayList<>();
+		for (DataBrowserEditor editor : getCurrentDataBrowsers()) {
+			editorNames.add(editor.getTitle());
+		}
+		return editorNames;
+	}
+	
+	private void addPVToEditor(String pvAddress, final String displayName, DataBrowserEditor editor) {
+		if (editor == null) {
+	        return;
+	    }
+		Model model = editor.getModel();
+		// Add received items
+	    final double period = Preferences.getScanPeriod();
+	    try {
+			final PVItem item = new PVItemWithUnits(displayName, pvAddress, period);
+			item.useDefaultArchiveDataSources();
+			// Add item to new axes
+			AxisConfig axis = model.addAxis(item.getDisplayName());
```

Comment:
This typically happens before `setUnits` on `item` is called via the observer, meaning that axis label is set without the EGU. Can you force `item` to initialize the EGU when instantiating?